src/compiler/crystal/semantic/top_level_visitor.cr

```diff
@@ -1049,6 +1049,12 @@ class Crystal::TopLevelVisitor < Crystal::SemanticVisitor
         node.returns_twice = true
       when @program.raises_annotation
         node.raises = true
+      when @program.deprecated_annotation
+        # Check a DeprecatedAnnotation can be built
```

Comment:
I'm not exactly sure what this comment means but perhaps
```suggestion
        # Check whether the DeprecatedAnnotation can be built.
```

Reply:
It means to check the arguments are the expected ones. Annotations are a bit loosely defined. There is currently no way to declare which arguments are expected.